pytket/extensions/cutensornet/structured_state/mps.py

```diff
@@ -288,6 +288,75 @@ def apply_qubit_relabelling(self, qubit_map: dict[Qubit, Qubit]) -> MPS:
         self._logger.debug(f"Relabelled qubits... {qubit_map}")
         return self
 
+    def add_qubit(self, new_qubit: Qubit, position: int, state: int = 0) -> MPS:
+        """Adds a qubit at the specified position.
+
+        Args:
+            new_qubit: The identifier of the qubit to be added to the state.
+            position: The location the new qubit should be inserted at in the MPS.
+                Qubits on this and later indexed have their position shifted by 1.
+            state: Choose either ``0`` or ``1`` for the new qubit's state.
+                Defaults to ``0``.
+
+        Returns:
+            ``self``, to allow for method chaining.
+
+        Raises:
+            ValueError: If ``new_qubit`` already exists in the state.
+            ValueError: If ``position`` is negative or larger than ``len(self)``.
+            ValueError: If ``state`` is not ``0`` or ``1``.
+        """
+        options = {"handle": self._lib.handle, "device_id": self._lib.device_id}
+
+        if new_qubit in self.qubit_position.keys():
+            raise ValueError(
+                f"Qubit {new_qubit} cannot be added, it already is in the MPS."
+            )
+        if position < 0 or position > len(self):
+            raise ValueError(f"Index {position} is not a valid position in the MPS.")
+        if state not in [0, 1]:
+            raise ValueError(
+                f"Cannot initialise qubit to state {state}. Only 0 or 1 are supported."
+            )
+
+        # Identify the dimension of the virtual bond where the new qubit will appear
+        if position == len(self):
+            dim = self.get_virtual_dimensions(len(self) - 1)[1]  # Rightmost bond
+        else:  # Otherwise, pick the left bond of the tensor currently in ``position``
+            dim = self.get_virtual_dimensions(position)[0]
+
+        # Create the tensor for I \otimes |state>
+        identity = cp.eye(dim, dtype=self._cfg._complex_t)
+        qubit_tensor = cp.zeros(2, dtype=self._cfg._complex_t)
+        qubit_tensor[state] = 1

```

Comment:
For reference: this is the line that initialises the state at either |0> or |1> depending of the value of `state`. Here, `qubit_tensor` is just the vector [1 0] for |0> or [0 1] for |1>.